integration-tests/redis_test.go

```diff
@@ -21,19 +21,44 @@ const (
 
 var customRedisPlans = []map[string]any{
 	customRedisPlan,
+	redisPlanWithFlexibleNodeType,
+	deprecatedCacheSizePlan,
 }
 
 var customRedisPlan = map[string]any{
 	"name":        redisCustomPlanName,
 	"id":          redisCustomPlanID,
 	"description": "Beta - Default Redis plan",
-	"cache_size":  2,
```

Comment:
question: this wouldn't be possible without the new computed property, am I wrong?
I can't understand why this is possible and it passes the validation when registering the plan

Reply:
With this PR `cache_size` [is no longer required](https://github.com/cloudfoundry/csb-brokerpak-aws/pull/774/files#diff-3c895c4ef9e3489a36f4c02737ffd079a7e4e92a4e998525c114e74134840fdcL29).

So in theory, not specifying `cache_size` in the plan is perfectly legitimate without the computed property.
However, [the logic for calculating the `node_type` based on `cache_size` has been maintained for backward compatibility](https://github.com/cloudfoundry/csb-brokerpak-aws/blob/8c2fec1601d7c1ee863de778a79706f4103181d8/terraform/redis/cluster/provision/data.tf#L38).

Without the computed property making `cache_size` default to -1 (or any other arbitrary value), this expression would fail: `node_type = length(var.node_type) == 0 ? local.instance_types[var.cache_size] : var.node_type` causing any attempt to create a new service without specifying `cache_size` to fail.

Reply:
I'm open for suggestions on how to implement this in a clearer way.
All the alternatives I could think of had their own tradeoff. For example:
An overly-complicated logic for calculating `node_type` based on `cache_size` which take into account when `cache_size` is null.

I tried to justify the approach I finally took in this comment:
https://github.com/cloudfoundry/csb-brokerpak-aws/blob/8c2fec1601d7c1ee863de778a79706f4103181d8/terraform/redis/cluster/provision/data.tf#L22-L25